### PR TITLE
Refactor FXIOS-5842 [v112] fix testable import client warnings

### PR DESCRIPTION
**Pull request by petruSt24**
https://mozilla-hub.atlassian.net/browse/FXIOS-5842
The test module is implicitly importing the main module's bridging header. You don't need to import it. Everything is OK.
Everything is OK now, but it won't be OK later because this "implicit" import will disappear soon (deprecated) and thus, it's not gonna be OK anymore.
In order to stay OK, you should "explicitly" import your main module's bridging header in your test module's bridging header.

### CODE REVIEW

**Comment by petruSt24** [comment]
https://mozilla-hub.atlassian.net/browse/FXIOS-5842 The test module is implicitly importing the main module's bridging header. You don't need to import it. Everything is OK.
Everything is OK now, but it won't be OK later because this "implicit" import will disappear soon (deprecated) and thus, it's not gonna be OK anymore.
In order to stay OK, you should "explicitly" import your main module's bridging header in your test module's bridging header.

File: Client/Frontend/Browser/TabManagement/Tab+ChangeUserAgent.swift [hunk 1]
```diff
@@ -13,7 +13,8 @@ extension Tab {
         }()
 
         private static var baseDomainList: Set<String> = {
-            if let hosts = NSKeyedUnarchiver.unarchiveObject(withFile: ChangeUserAgent.file.path) as? Set<String> {
+            if let data = try? Data(contentsOf: ChangeUserAgent.file),
+               let hosts = try? NSKeyedUnarchiver.unarchivedObject(ofClasses: [NSSet.self, NSArray.self, NSString.self], from: data) as? Set<String> {
```
Comment (lmarceau) [inline Client/Frontend/Browser/TabManagement/Tab+ChangeUserAgent.swift:17]: Oh those changes are from the other task, they shouldn't be on this branch 🤔

File: Tests/ClientTests/Frontend/Browser/TabManagement/Tab+ChangeUserAgentTests.swift [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+@testable import Client
+
+import XCTest
+
+class ChangeUserAgentTests: XCTestCase {
```
Comment (lmarceau) [inline Tests/ClientTests/Frontend/Browser/TabManagement/Tab+ChangeUserAgentTests.swift:9]: Same, those changes shouldn't be on this branch

File: Client/Frontend/Browser/TabManagement/Tab+ChangeUserAgent.swift (hunk 1 above)
Comment (petruSt24) [reply Client/Frontend/Browser/TabManagement/Tab+ChangeUserAgent.swift:17]: damn, I forgot to switch to main before creating this branch, should I try to rebase the branch to main or delete it and recreate the changes from main?

Comment (lmarceau) [reply Client/Frontend/Browser/TabManagement/Tab+ChangeUserAgent.swift:17]: As you wish! You can try rebasing, worst case you can close this PR and open a new one by cherry picking the commits for this work

**Comment by lmarceau** [comment]
Closing in favor of https://github.com/mozilla-mobile/firefox-ios/pull/13574

**Pull request closed without merging**